Plugins/Traccc/include/Acts/Plugins/Traccc/MeasurementConversion.hpp

```diff
@@ -0,0 +1,164 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+// Plugin include(s)
+#include "Acts/Plugins/Traccc/Detail/AlgebraConversion.hpp"
+
+// Acts include(s)
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Definitions/TrackParametrization.hpp"
+#include "Acts/EventData/Measurement.hpp"
+#include "Acts/EventData/SourceLink.hpp"
+#include "Acts/Geometry/GeometryIdentifier.hpp"
+
+// Detray include(s)
+#include "detray/core/detector.hpp"
+#include "detray/tracks/bound_track_parameters.hpp"
+
+// Traccc include(s)
+#include "traccc/definitions/qualifiers.hpp"
+#include "traccc/definitions/track_parametrization.hpp"
+#include "traccc/edm/measurement.hpp"
+#include "traccc/edm/track_state.hpp"
+
+// System include(s)
+#include <memory>
+#include <variant>
+
+namespace Acts::TracccPlugin {
+
+/// @brief Converts a traccc bound index to an Acts bound index.
+/// @param tracccBoundIndex the traccc bound index.
+/// @returns an Acts bound index.
+inline Acts::BoundIndices boundIndex(
+    const traccc::bound_indices tracccBoundIndex) {
+  switch (tracccBoundIndex) {
+    case traccc::bound_indices::e_bound_loc0:
+      return Acts::BoundIndices::eBoundLoc0;
+    case traccc::bound_indices::e_bound_loc1:
+      return Acts::BoundIndices::eBoundLoc1;
+    case traccc::bound_indices::e_bound_phi:
+      return Acts::BoundIndices::eBoundPhi;
+    case traccc::bound_indices::e_bound_theta:
+      return Acts::BoundIndices::eBoundTheta;
+    case traccc::bound_indices::e_bound_qoverp:
+      return Acts::BoundIndices::eBoundQOverP;
+    case traccc::bound_indices::e_bound_time:
+      return Acts::BoundIndices::eBoundTime;
+    case traccc::bound_indices::e_bound_size:
+      return Acts::BoundIndices::eBoundSize;
+    default:
+      throw std::runtime_error("Could not convert traccc bound index");
+  }
+}
+
+/// @brief Creates an Acts measurement from a traccc measurement.
+/// @tparam the dimension of the Acts measurement (subspace size).
+/// @param m the traccc measurement.
+/// @param sl the Acts source link to use for the Acts measurement.
+/// @returns an Acts measurement with data copied from the traccc measurement
+/// and with its source link set to the one provided to the function.
+template <std::size_t dim>
+inline Acts::Measurement<Acts::BoundIndices, dim> measurement(
```

Comment:
What's the plan for this? @andiwand is phasing out `Acts::Measuement` from core and pushing it into the Examples (#3282), so this would have to be removed from the plugin fairly soon.